irods/manager/data_object_manager.py

```diff
@@ -138,8 +138,11 @@ def chksum(self, path, **options):
         with self.sess.pool.get_connection() as conn:
             conn.send(message)
             response = conn.recv()
-            results = response.get_main_message(DataObjChksumResponse)
-            checksum = results.myStr
+            try:
+                results = response.get_main_message(DataObjChksumResponse)
+                checksum = results.myStr
+            except iRODSMessage.ResponseNotParseable:
```

Comment:
consider a comment here that explains this means the data object / replica didn't have a replica in the catalog.

Reply:
Commented it, but actually this means that VERIFY_CHKSUM_KW keyword was used, in which case the server does not return (or store) a chksum string.

I think semantically this ties in with the RErrorMsg #287  thing... each replica with a missing or incorrect chksum would then have its own entry in that message array, identifying it by replNum and stating it is wrong / missing / etc.

Reply:
right, got it.